File: python/fundamentals/day-01/problem4.py
# Program to calculate the area of a circle using π ≈ 3.14159

def calculate_circle_area(radius):
    """Calculate area of a circle given the radius."""
    PI = 3.14159
    return PI * (radius ** 2)

def main():
    try:
        # Take radius input from user
        radius_input = input("Enter the radius of the circle: ").strip()
        
        # Convert to float
        radius = float(radius_input)
        
        # Validate radius
        if radius < 0:
            print("Error: Radius cannot be negative.")
            return
        
        # Calculate and display area
        area = calculate_circle_area(radius)
        print(f"Area of the circle with radius {radius} is: {area:.4f}")
    
    except ValueError:
        print("Invalid input. Please enter a numeric value for the radius.")

if __name__ == "__main__":
    main()

File: java/fundamentals/day01/problem4.java
package fundamentals.day01;
import java.util.Scanner;
public class problem4 {
	public static void main(String[] args) {
        final double PI = 3.14159; // Approximation of π
        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Enter the radius of the circle: ");
            if (!scanner.hasNextDouble()) {
                System.out.println("Invalid input. Please enter a numeric value.");
                return;
            }

            double radius = scanner.nextDouble();

            // Validate radius
            if (radius < 0) {
                System.out.println("Radius cannot be negative.");
                return;
            }

            // Calculate area
            double area = PI * radius * radius;

            // Display result with 5 decimal places
            System.out.printf("Area of the circle: %.5f%n", area);

        } catch (Exception e) {
            System.out.println("An unexpected error occurred: " + e.getMessage());
        } finally {
            scanner.close();
        }
    }
}

